wordpress-mobile/WordPress-iOS
# Jetpack Social: Add icon download for social media icons that aren't bundled

See: #20784

## Description

Adds an icon download to the no connections view and matches the `PublicizeService` objects that are cached for the blog.

## Screenshots

| Light | Dark |
|------|------|
| ![Screenshot Light](https://github.com/wordpress-mobile/WordPress-iOS/assets/2454408/4ed6ae2c-2c79-40db-90b9-a4ea0af24918) | ![Screenshot Dark](https://github.com/wordpress-mobile/WordPress-iOS/assets/2454408/344da7c2-730c-4bd1-882b-df59968db456) |

## Testing

To test:
- Launch Jetpack and login
- Select a site which has no social connections but has services available
- Ensure the `PublicizeService` objects are cached (this can be done by going to the social screen)
- Create a blog post or edit one
- On the editor, open the menu (`...`) in the top right
- Tap on `Post Settings`
- **Verify** the icons match the services
- Tap back
- Disable your internet
- Open the `Post Settings` again
- **Verify** the placeholder is used for icons on services that we don't bundle an icon for

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:


UI Changes testing checklist:
